src/features/encoders/wp2/worker/wp2Encode.ts

```diff
@@ -10,24 +10,56 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-import wp2Encoder, { WP2Module } from 'codecs/wp2/enc/wp2_enc';
-import wasmUrl from 'url:codecs/wp2/enc/wp2_enc.wasm';
-import { initEmscriptenModule } from 'features/worker-utils';
+import type { WP2Module } from 'codecs/wp2/enc/wp2_enc';
 import type { EncodeOptions } from '../shared/meta';
 
+import { initEmscriptenModule } from 'features/worker-utils';
+import { threads, simd } from 'wasm-feature-detect';
+
+import wasmUrl from 'url:codecs/wp2/enc/wp2_enc.wasm';
+
+import wasmUrlWithMT from 'url:codecs/wp2/enc/wp2_enc_mt.wasm';
+import workerUrl from 'omt:codecs/wp2/enc/wp2_enc_mt.worker.js';
+
+import wasmUrlWithMTAndSIMD from 'url:codecs/wp2/enc/wp2_enc_mt_simd.wasm';
+import workerUrlWithSIMD from 'omt:codecs/wp2/enc/wp2_enc_mt_simd.worker.js';
+
 let emscriptenModule: Promise<WP2Module>;
 
+async function init() {
+  if (await threads()) {
```

Comment:
```suggestion
  // At the time of writing, all browsers that support Wasm SIMD also support Wasm threads.
  // We might have to revisit this logic in the future.
  if (await threads()) {
```

Reply:
Will we add this to every codec using threads+SIMD or better to document this elsewhere in a centralised way if we want to?

Reply:
For now, I’d just keep adding this comment until we have a better solution. Open to suggestions.

Reply:
I personally don't think it needs explanation, it's just "a combo we currently support" :)

Reply:
Initially I thought it would require explanations if we used just `if (await simd()) { ...assume that threads are also available... }` like initially discussed, but we're not relying on this kind of magic here, so the conditions on their own seem fairly straightforward to follow.